Changelog.md

```diff
@@ -1,26 +1,39 @@
 # Change Log
 ## bpaf [0.6.0] - unreleased
-- removed `positional_os()` and `argument_os()`
-- removed `from_str`
-- internal refactors
-- documentation refactor: examples with input/output
-- `any` - a positional way to capture anything
-- `adjacent` restriction
-- `catch` inside `ParseOption`, `ParseMany` and `ParseSome`
-- usage line refactor, cosmetic mostly
-- include \n in --version, cosmetic
-
-Migration guide:
-1. replace `positional_os("FOO")` with `positional::<Type>("FOO")` or use `positional` in derive API
-2. replace `argument_os("FOO")` with `argument::<Type>("FOO")` or use `argument` in derive API
-3. replace any uses of `from_str::<T>` with turbofish on the nearest `positional` or
-   `argument`. You can use `map` to apply `FromStr` parser if `String` gets created inside your parsers
-
-You shouldn't be using those names directly, but there are some structure renames:
-1. `Positional` -> `ParsePositional`
-2. `BuildArgument` -> ParseArgument`
-3. `Command` -> `ParseCommand`
-
+# What's new in 0.6.0
+- `adjacent` restriction to parse things in a tighter context
+- `catch` for `many`, `some` and `optional` to handle parse errors
+- `any` positional like, to consume pretty much anything from a command line
+- improved documentation with more detailed examples
+- cosmetic improvements
+- a sneaky reminder to use `to_options` on `Parser` before trying to run it
+- a way to make boxed parsers with single item `construct!` macro for making dynamic parsers
+- a horrible way to reduce Yoda-talk coding by placing primitive definitions inside the `construct!` macro
+
+With new additions you should be able to parse pretty much anything and then some more :)
+
+# Migration guide 0.5.x -> 0.6.x
+
```

Comment:
> One way would be to rename OptionParser into Parser - now that Parser is a trait they can coexist, and with hidden deprecated .run() in Parser trait to serve as a reminder - it should be harder to confuse those two I think. to_options is still a thing, with OptionParser renamed it can probably be changed to done/finish/make or add_info/with_info/and_info?

I would recommend against `Parser` and `Parser` as that can be confusing even if they do operate in different namespaces.

As for `to_options`:
- `done`, et all implies it is built but it isn't
- `add_info` focuses purely on adding context but there is more going on

I would focus on the core of what it is.  It is taking the lower level parsers and wrapping / building / turning them into a full application / user-facing / std::env::args parser.

Reply:
You also go though `OptionParser` when implementing commands and subcommands, making tests for your code or parsing things from sources other than `std::env::args_os`. I think of it as a parser with some extra meta information attached. Interfacing with the lower level is done inside the `run` method only - and that's the part that needs to be pluggable in order to support windows style flags FWIW.